dapps/marketplace/src/pages/user/Profile.js

```diff
@@ -131,25 +124,38 @@ class UserProfile extends Component {
     clearTimeout(this.timeout)
   }
 
-  changesPublishedToBlockchain(props, prevProps) {
+  changesPublishedToBlockchain(props, prevProps, state, prevState) {
     const profile = get(props, 'identity') || {}
     const prevProfile = get(prevProps, 'identity') || {}
 
+    const verifiedAttestations = state.mappedVerifiedAttestations
+    const prevVerifiedAttestations = prevState.mappedVerifiedAttestations
+
     return (
       (profile.firstName !== prevProfile.firstName ||
         profile.lastName !== prevProfile.lastName ||
         profile.description !== prevProfile.description ||
         profile.avatarUrl !== prevProfile.avatarUrl ||
-        profile.emailVerified !== prevProfile.emailVerified ||
-        profile.phoneVerified !== prevProfile.phoneVerified ||
-        profile.facebookVerified !== prevProfile.facebookVerified ||
-        profile.googleVerified !== prevProfile.googleVerified ||
-        profile.twitterVerified !== prevProfile.twitterVerified ||
-        profile.airbnbVerified !== prevProfile.airbnbVerified ||
-        profile.websiteVerified !== prevProfile.websiteVerified ||
-        profile.kakaoVerified !== prevProfile.kakaoVerified ||
-        profile.githubVerified !== prevProfile.githubVerified ||
-        profile.linkedinVerified !== prevProfile.linkedinVerified) &&
+        verifiedAttestations.emailVerified !==
```

Comment:
Can we just check if any keys are different instead of each on individually?

Reply:
Yep, an array diff of object keys should do. Will update it.